src/ImageSharp/Formats/Jpeg/Components/Decoder/JpegBitReader.cs

```diff
@@ -212,7 +212,7 @@ public bool FindNextMarker()
     private int ReadStream()
     {
         int value = this.badData ? 0 : this.stream.ReadByte();
-        if (value == -1)
+        if (value == -1 || this.stream.Position >= this.stream.Length)
```

Comment:
>Seeking to any location beyond the length of the stream is supported.

https://learn.microsoft.com/en-us/dotnet/api/system.io.stream.position?view=net-7.0#remarks

Reply:
Turns out the fix is bad. It's somehow preventing complete data reading.

Reply:
Proper fix implemented now.